yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -150,18 +153,21 @@ export class AztecNodeService implements AztecNode {
     // this may well change in future
     config.transactionProtocol = `/aztec/tx/${config.l1Contracts.rollupAddress.toString()}`;
 
+    // now create the merkle trees and the world state synchronizer
+    const worldStateSynchronizer = await createWorldStateSynchronizer(config, store, archiver, telemetry);
+
     // create the tx pool and the p2p client, which will need the l2 block source
     const p2pClient = await createP2PClient(
       config,
       store,
       new AztecKVTxPool(store, telemetry),
       new InMemoryAttestationPool(),
       archiver,
+      new GlobalVariableBuilder(config),
+      new TestCircuitVerifier(),
+      worldStateSynchronizer,
```

Comment:
Rather than passing all the deps for the TxValidators across the p2p client and lib p2p service, and creating a new TxValidator for every tx, I'd change it so this call accepts a TxValidator (or a TxValidatorFactory if needed) directly. So when we add new validations with new deps we don't need to keep changing the entire call chain.

Reply:
I started writing it this way actually!

Issue is we sort of need to know which specific validation fails or not in order to then update peer scores based on the result